### PR TITLE
Hides error textview in the header for textInputLayout views.

### LINKED ISSUE #1725: Add/Edit patient screen UI broken in demo app
**Describe the bug**
Add/Edit patient screen UI broken in demo app, there gap between UI item is increased

**Component**
SDC library, rSDC gallery app

**To Reproduce**
1.. Launch'Demo'
2. Click on '+' to add new patient / select any patient from patient list to and click on edit
3. Scroll up and down
4. See error : 1. Space is more in label and the edit box for same label eg: First Name and its EditText box, Phone Number and its EditText box, etc 2. Scroll at the bottom of screen, there is no margin in parent bottom and last Edit text.

**Expected behavior**
there should be proper margin in Ui components on screen
**Screenshots**
[AddPatient_or_EditPatient_screenBroken.webm](https://user-images.githubusercontent.com/13957417/202093426-77ab1d2c-b178-4923-9f21-ff46d3913066.webm)

**Smartphone (please complete the following information):**
 - Device: Pixel5 emulator
 - Android version: API 32

**Additional context**

**Would you like to work on the issue?**
Please state if this issue should be assigned to you or who you think could help to solve this issue.

### PR DESCRIPTION
**Pull request by santosh-pingle**
**IMPORTANT: All PRs must be linked to an issue (except for extremely trivial and straightforward changes).**

Fixes #1725

**Description**
Error text view in the header was not hidden for views which shows error text at the bottom of the view. As result it was always showing extra blank space in the header.
Now, error text view in the header is hidden, therefore it will not add extra blank space.

**Alternative(s) considered**
Have you considered any alternatives? And if so, why have you chosen the approach in this PR?

**Type**
Choose one: Bug fix 

**Screenshots (if applicable)**

**Checklist**
- [x] I have read and acknowledged the [Code of conduct](https://github.com/google/android-fhir/blob/master/docs/CODE_OF_CONDUCT.md).
- [x] I have read the [Contributing](https://github.com/google/android-fhir/wiki/Contributing) page.
- [x] I have signed the Google [Individual CLA](https://cla.developers.google.com/about/google-individual), or I am covered by my company's [Corporate CLA](https://cla.developers.google.com/about/google-corporate ).
- [x] I have discussed my proposed solution with code owners in the linked issue(s) and we have agreed upon the general approach.
- [x] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the style guide of this project.
- [x] I have run `./gradlew check` and `./gradlew connectedCheck` to test my changes locally.
- [x] I have built and run the demo app(s) to verify my change fixes the issue and/or does not break the demo app(s).

### CODE REVIEW

**Comment by santosh-pingle** [comment]
Screen Shot 2022-11-22 at 12 17 46 PM

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAutoCompleteViewHolderFactoryTest.kt [hunk 1]
```diff
@@ -228,6 +228,21 @@ class QuestionnaireItemAutoCompleteViewHolderFactoryTest {
       .isNull()
   }
 
+  @Test
+  fun `hides error textview in the header`() {
```
Comment (PallaviGanorkar) [inline datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAutoCompleteViewHolderFactoryTest.kt:232]: If possible you can add similar test case in classes like QuestionnaireItemDialogSelectViewHolderFactory.kt
QuestionnaireItemDisplayViewHolderFactory
QuestionnaireItemSimpleQuestionAnswerDisplayViewHolderFactory

**Review by PallaviGanorkar — COMMENTED** [review]
@santosh-pingle Thanks!
Added one comment, other LGTM

File: datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAutoCompleteViewHolderFactoryTest.kt (hunk 1 above)
Comment (santosh-pingle) [reply datacapture/src/test/java/com/google/android/fhir/datacapture/views/QuestionnaireItemAutoCompleteViewHolderFactoryTest.kt:232]: done.

File: datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest.kt [hunk 2]
```diff
@@ -348,6 +351,23 @@ class QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest {
     onView(withId(R.id.add_another)).check(ViewAssertions.matches(ViewMatchers.isDisplayed()))
   }
 
+  @Test
+  fun `errorTextviewInHeader_shouldHide`() {
```
Comment (jingtang10) [inline datacapture/src/androidTest/java/com/google/android/fhir/datacapture/views/QuestionnaireItemDialogMultiSelectViewHolderFactoryEspressoTest.kt:355]:
```suggestion
  fun `shouldHideErrorTextviewInHeader`() {
```

**Review by jingtang10 — APPROVED** [review]
Thanks @santosh-pingle

**Pull request merged**